cmd/argocd/commands/app.go

```diff
@@ -1539,6 +1540,7 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
+			output, _ = c.Flags().GetString("output")
```

Comment:
I don't think this is needed. The `&output` reference below will make sure the `output` value is updated correctly.